qiskit/providers/aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,204 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Statevector quantum state class.
+"""
+import copy
+import numpy as np
+
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.providers.aer.quantum_info.states.aer_state import AerState
+
+
+class AerStatevector(Statevector):
+    """AerStatevector class"""
+
+    def __init__(self, data, **configs):
+        """Initialize a statevector object.
+
+        Args:
+            data (np.array or list or Statevector or QuantumCircuit or qiskit.circuit.Instruction):
+                Data from which the statevector can be constructed. This can be either a complex
+                vector, another statevector or a ``QuantumCircuit`` or ``Instruction``
+                (``Operator`` is not supportted in the current implementation).  If the data is
+                a circuit or instruction, the statevector is constructed by assuming that all
+                qubits are initialized to the zero state.
+            configs (kwargs): configurations of ```AerState`. All the keys are handeld as string
+
+        Raises:
+            QiskitError: if input data is not valid.
+
+        """
+        if '_aer_state' in configs:
+            self._aer_state = configs.pop('_aer_state')
+        if isinstance(data, (QuantumCircuit, Instruction)):
+            data, aer_state = AerStatevector._from_instruction(data, None, **configs)
+            self._aer_state = aer_state
+        elif isinstance(data, np.ndarray):
+            data, aer_state = AerStatevector._from_ndarray(data, **configs)
+            self._aer_state = aer_state
+        else:
+            self._aer_state = None
+        super().__init__(data)
+        self._result = None
+        self._configs = configs
+
+    def aer(self):
+        return self._aer_state is not None
+
+    def _assert_aer_mode(self):
+        if not self.aer():
+            raise AerError('AerState was not used.')
+
+    def _last_result(self):
+        self._assert_aer_mode()
+        if self._result is None:
+            self._result = self._aer_state.last_result()
+        return self._result
+
+    def _metadata(self):
+        self._assert_aer_mode()
+        if self._last_result() is None:
+            raise AerError('AerState was not used and metdata does not exist.')
+        return self._last_result()['metadata']
+
+    @property
+    def method(self):
+        return self._metadata()['method']
+
+    @property
+    def device(self):
+        return self._metadata()['device']
+
+    @property
+    def fusion_config(self):
+        return self._metadata()['fusion']
+
+    @property
+    def parallel_state_update(self):
+        return self._metadata()['parallel_state_update']
```

Comment:
These also need docstrings so users know what they are.

Also why are these options hard coded but not other ones?